Feature GUI

rryter opened the pull request:
Very simple GUI to start and monitor an Archive- or Validator-Node with a few clicks.
## General
- [ ] Add GET /health endpoint

## Archive Node
- [x] Start Node with a click
- [x] Monitoring Vanguard & Pandora (Peers and Latest Block)
- [ ] Monitoring Orchestrator

## Validator Node
- [x] Generating Keys with a click on a Button (wip)
- [x] Importing Keys (todo)
- [x] Start and stop a potentially running node and run it as a validator
- [x] Monitoring Validator

## Design
- [x] Decide on which Design to use.

rryter committed:
chore(setup): initial commit

rryter committed:
chore(setup): initial mvp

rryter committed:
chore(typo): fix it

rryter committed:
Update README.md

rryter committed:
chore(status): remove metrics search for now

rryter committed:
chore(status): make it right

rryter committed:
chore(vanguard): remove pending validator

rryter committed:
chore(vanguard): remove pending validator

rryter committed:
chore(build): avoid cache issue

rryter committed:
Update README.md

rryter committed:
Update README.md

rryter committed:
chore(styling): made it prettier, added info

rryter committed:
Merge branch 'main' of https://github.com/rryter/node-status into main

rryter committed:
chore(styles): remove unnescessary code

rryter committed:
chore(colors): fix

rryter committed:
chore(setup): suitable architecture

rryter committed:
chore(setup): wip

rryter committed:
Merge branch 'main' of https://github.com/lukso-network/network-lukso-cli into feature/gui

rryter committed:
chore(go): small improvements

rryter committed:
chore(readme): update info

rryter committed:
chore(readability): proper variable names

rryter committed:
chore(tests): fix tests, to be ready to test

rryter committed:
chore(gh-action): add test step

blazejkrzak commented on apps/lukso-manager/src/downloader/main.go:
```diff
@@ -0,0 +1,270 @@
+		panic(err1)
+	}
+
+	// data, _ := ioutil.ReadFile(filePath)
```
do not leave commented code

blazejkrzak commented on apps/lukso-manager/src/downloader/main.go:
```diff
@@ -0,0 +1,270 @@
+	var t updateRequestBody
+	err := decoder.Decode(&t)
+	if err != nil {
+		panic(err)
```
you mix panics with fatal logs

blazejkrzak commented on apps/lukso-manager/src/main.go:
```diff
@@ -0,0 +1,64 @@
+	"github.com/gorilla/mux"
+)
+
+func main() {
```
do GET `/health` as well, so we remove this exec command from gke

rryter replied:
Good Idea.

- [ ] Add GET /health endpoint

blazejkrzak commented on apps/lukso-manager/src/metrics/main.go:
```diff
@@ -0,0 +1,35 @@
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		log.Fatalln(err)
```
Idk if pattern with fatal deaths is good for community.
Other thing is that you do not close response body with `defer`

blazejkrzak commented on apps/lukso-manager/src/runner/config.go:
```diff
@@ -0,0 +1,33 @@
+)
+
+type NetworkConfig struct {
+	GENESISTIME       int    `yaml:"GENESIS_TIME"`
```
lets do `genesisTime` and consecutively with others

rryter replied:
```suggestion
	genesisTime       int    `yaml:"GENESIS_TIME"`
```

Are you sugessting this?

rryter committed:
chore(ci): run tests

blazejkrzak commented on apps/lukso-manager/src/runner/config.go:
```diff
@@ -0,0 +1,33 @@
+
+func ReadConfig(network string) (*NetworkConfig, error) {
+	fileName := "/opt/lukso/networks/" + network + "/config/network-config.yaml"
+	buf, err := ioutil.ReadFile(fileName)
```
defer buf.close()

rryter replied:
<img width="905" alt="Screenshot 2021-11-08 at 15 21 01" src="https://user-images.githubusercontent.com/798709/140758485-8a2ef132-2016-4dbe-a895-3c6af0ee02df.png">

I suppose the variable naming was bad.

blazejkrzak commented on apps/lukso-manager/src/runner/pandora.go:
```diff
@@ -0,0 +1,77 @@
+		"--ethstats=l15-rryter-gui:6Tcpc53R5V763Aur9LgD@" + statsPrefix + "stats.pandora.l15.lukso.network",
+	}
+
+	fmt.Println("/home/rryter/.lukso/downloads/" + client + "/" + version + "/" + client + " --datadir " + datadir + " init /opt/lukso/networks/" + network + "/config/pandora-genesis.json &>/dev/null")
```
dehardcode

blazejkrzak commented on apps/lukso-manager/src/runner/pandora.go:
```diff
@@ -0,0 +1,77 @@
+		log.Fatal(startError)
+	}
+
+	cmnd.Wait()
```
its common to use `cmd` or `command`

blazejkrzak commented on apps/lukso-manager/src/validator/main.go:
```diff
@@ -0,0 +1,71 @@
+	entropy, _ := bip39.NewEntropy(256)
+	mnemonic, _ := bip39.NewMnemonic(entropy)
+
+	// Generate a Bip32 HD wallet for the mnemonic and a user supplied password
```
do not leave commented code

blazejkrzak commented on apps/lukso-manager/src/validator/main.go:
```diff
@@ -0,0 +1,71 @@
+	// Generate a Bip32 HD wallet for the mnemonic and a user supplied password
+	// seed := bip39.NewSeed(mnemonic, password)
+
+	folder := "/home/rryter/.lukso/networks/" + body.Network
```
I believe that you can handle it in more dynamic approach :)

rryter replied:
Is it too much to ask of people to create an `rryter` folder? come on. 😂

blazejkrzak commented on apps/lukso-manager/src/webserver/dist/apps/lukso-status/main.6fd04046714fcaa29b9b.js:
```diff
@@ -0,0 +1 @@
+(self.webpackChunklukso=self.webpackChunklukso||[]).push([[179],{82:Jr=>{function un(Yr){return Promise.resolve().then(()=>{var cn=new Error("Cannot find module '"+Yr+"'");throw cn.code="MODULE_NOT_FOUND",cn})}un.keys=()=>[],un.resolve=un,un.id=82,Jr.exports=un},621:(Jr,un,Yr)=>{"use strict";function cn(t){return"function"==typeof t}let Xi=!1;const mt={Promise:void 0,set useDeprecatedSynchronousE ...
```
IDK if we should track by git this path `apps/lukso-manager/src/webserver/dist/***`

rryter replied:
No we should not, it's fixed.

blazejkrzak commented on apps/lukso-manager/src/webserver/main.go:
```diff
@@ -0,0 +1,33 @@
+}
+
+func sendErr(w http.ResponseWriter, code int, message string) {
+	resp, _ := json.Marshal(map[string]string{"error": message})
```
potential risk here that we return errors without sanitization

blazejkrzak commented on apps/lukso-status-e2e/src/support/commands.ts:
```diff
@@ -0,0 +1,33 @@
+});
+//
+// -- This is a child command --
+// Cypress.Commands.add("drag", { prevSubject: 'element'}, (subject, options) => { ... })
```
is this necessary to have code comments here?

rryter replied:
That's auto generated, I'll clean this up as soon as we have a working product, I'll be writing E2E tests for this, so I will come to this section and clean it up

blazejkrzak commented on apps/lukso-status/src/app/components/status/orchestrator-status/orchestrator-status.component.html:
```diff
@@ -0,0 +1,22 @@
+  <h4 class="card-header">
+    Orchestrator
+    <span class="badge bg-warning">Ok</span>
+    <!-- <ng-container *ngIf="metrics[peersSelector] as peers">
```
Commented code here

blazejkrzak commented on apps/lukso-status/src/app/modules/launchpad/launchpad/components/launchpad.component.ts:
```diff
@@ -0,0 +1,45 @@
+        .subscribe(() => {
+          this.isGeneratingKeys = false;
+        });
+    } else {
```
do not do else statements, they are not needed. Use `early returns`

blazejkrzak commented on apps/lukso-status/src/app/modules/launchpad/launchpad/services/keygen.service.ts:
```diff
@@ -0,0 +1,20 @@
+  }
+
+  genereateKeys(password: string, network: string, amountOfValidators: string) {
+    return this.httpClient.post('/api/launchpad/generate-keys', {
```
I think that entire setup could be handled via WS instead of REST, but to simplify V1 alpha release lets stay with REST-like.
BTW you can create const/enum for endpoints

rryter replied:
Yes I agree, the goal should be websockets. Once we have a working product, we should change this.

blazejkrzak commented on apps/lukso-status/src/app/services/vanguard-metrics.service.ts:
```diff
@@ -0,0 +1,45 @@
+          .get('/vanguard/metrics', {
+            responseType: 'text',
+          })
+          .pipe(
```
have you ever heard of `async/await`? : )

rryter replied:
That's not applicable here, Angular uses RxJS which is way more powerful. 👍

blazejkrzak commented on unused-markup.md:
```diff
@@ -0,0 +1,131 @@
+ <!-- <tr>
```
what is the purpose of this?

rryter replied:
stuff I might need again later on, I can track it outside of git, i'll remove this file.

rryter committed:
chore(webserver): do not track generated files

rryter committed:
chore(cleanup): remove commented code

rryter committed:
chore(codestyle): remove obsolete else

rryter committed:
chore(cleanup): remove unused markup, rename cmnd to command

rryter committed:
chore(setup): broken package references

blazejkrzak committed:
Changed repository structure to match golang standards

blazejkrzak committed:
Changed repository structure to match golang standards

rryter committed:
Merge pull request #2 from blazejkrzak/feature/gui

Feature/gui

rryter committed:
Merge branch 'feature/gui' of https://github.com/rryter/node-status into feature/gui

rryter committed:
chore(setup): package renamings

rryter committed:
chore(pr-feedback): dehardcode some paths

rryter committed:
chore(errors): improve handling

rryter committed:
chore(go): improve error handling

rryter committed:
chore(defaults): proper defaults for components

rryter committed:
chore(go): error handling improved

rryter committed:
chore(settings): persistent settings via bolt db

rryter committed:
chore(keys): minimal validator flow

rryter committed:
chore(validators): deposit transactions

rryter committed:
chore(configs): download configs before every start

rryter committed:
chore(nav): font + spacing + active state

rryter committed:
chore(style): make it less ugly

rryter committed:
chore(navbar): move network switcher to top navbar

rryter committed:
chore(go): remove unneeded log

rryter committed:
chore(peersOverTime): wip ugly

rryter committed:
chore(peersOverTime): wip ugly

rryter committed:
chore(peersOverTime): wip ugly

rryter committed:
chore(peersOverTime): wip ugly

rryter committed:
chore(peersOverTime): wip ugly

rryter committed:
chore(peersOverTime): wip ugly

rryter committed:
chore(peersOverTime): pandora

rryter committed:
chore(styling): general improvements

rryter committed:
chore(debug): added debug config

rryter committed:
chore(stats): network stats via websockets

rryter committed:
chore(styling): improve little by little

rryter committed:
feat(status): vanguard peers over time

rryter committed:
chore(cleanup): small cleanup

rryter committed:
chore(styling): improved styling, added font

rryter committed:
chore(toggles): individual toggle for pandora and vanguard

rryter committed:
chore(logging): logging for vanguard

rryter committed:
chore(logs): grouped by genesis time

rryter committed:
chore(defaults): propper default network

rryter committed:
chore(state): started using rx-angular

rryter committed:
chore(settings): cleanup

rryter committed:
chore(network-status): loading indicators

rryter committed:
Update README.md

rryter committed:
chore(wip): versions

rryter committed:
Merge branch 'feature/gui' of github.com:rryter/network-lukso-cli into feature/gui

rryter committed:
chore(create-keys): state refactoring

rryter committed:
chore(logs): remove unnescessary logs

rryter committed:
chore(naming): settings instead of config

rryter committed:
chore(mkdir): use mkdirall to be sure to create missing folders

rryter committed:
chore(config): guard existence

rryter committed:
chore(settings): spinner

rryter committed:
chore(settings): spinner

luki3k5 committed:
Changes allowing for quick start on the project

Co-authored with @rryter

Changelog:

- README: more verbose explanation on setup and configuration. Also assumes no prior setup or installation of go watcher.
- added subscribe() function on component settings.component.ts
- added a check for oldConfig existance before calling GENESISTIME on it.
- Changed os.Mkdir function into os.MkdirAll which creates entire path of directories and subdirectories.
- Removed the check for "-rc" to work wider choice of tagged versions of dependencies.

luki3k5 committed:
Squashed commit of the following:

commit 4beb904611690eb321484c3bf8fcf239be763854
Author: Reto Ryter <reto@twy.gmbh>
Date:   Tue Nov 23 21:43:49 2021 +0000

    chore(settings): spinner

commit d1192e9d77641ebddb5fc302048e106d142f93fa
Author: Reto Ryter <reto@twy.gmbh>
Date:   Tue Nov 23 21:37:47 2021 +0000

    chore(settings): spinner

commit 67a2a6cf23969c49b3eea4091e54cfbd5274415b
Author: Reto Ryter <reto@twy.gmbh>
Date:   Tue Nov 23 19:50:08 2021 +0000

    chore(config): guard existence

commit 1bcd26e0639324a0f3b384943fb99875ce8a80dd
Author: Reto Ryter <reto@twy.gmbh>
Date:   Tue Nov 23 19:45:01 2021 +0000

    chore(mkdir): use mkdirall to be sure to create missing folders

commit 22d814e0df4605ea1025801944362d4cc4a7c826
Author: Reto Ryter <reto@twy.gmbh>
Date:   Tue Nov 23 19:34:07 2021 +0000

    chore(naming): settings instead of config

commit 00a8beac7d59617c44774ccb77129f0774a55e29
Author: Reto Ryter <reto@twy.gmbh>
Date:   Tue Nov 23 17:51:46 2021 +0000

    chore(logs): remove unnescessary logs

commit ea8c0e5d21f9f4b9440e8f99e454c243acbeef3e
Author: Reto Ryter <reto@twy.gmbh>
Date:   Tue Nov 23 17:48:27 2021 +0000

    chore(create-keys): state refactoring

luki3k5 committed:
Automated OS discovery for proper binaries download

luki3k5 committed:
Update to README

Extra step for building the lukso-status added

rryter committed:
chore(setup): wip

rryter committed:
chore(peersOverTime): remove debug output

rryter committed:
chore(settings): fix version dropdowns

rryter committed:
chore(updates): suspense from rx-angular

rryter committed:
chore(setup): install screen

rryter committed:
chore(build): gui

luki3k5 committed:
Merge branch 'feature/gui' of github.com:rryter/network-lukso-cli into LL_setup_adjustments

rryter committed:
chore(version): 0.0.1-alpha.1

rryter committed:
chore(build): use yarn instead of npm

rryter committed:
chore(build): fix path

rryter committed:
chore(tests): disable until experimentation phase has ended

rryter committed:
chore(naming): renamed package

luki3k5 committed:
changes for OS discovery for binaries download

luki3k5 committed:
Merge branch 'feature/gui' of github.com:rryter/network-lukso-cli into LL_setup_adjustments

rryter committed:
Merge pull request #1 from luki3k5/LL_setup_adjustments

Ll setup adjustments

rryter committed:
chore(setup): redirect to setup page if no binaries available

rryter committed:
chore(cleanup): cleanup guard

rryter committed:
chore(setup): proper guard

rryter committed:
chore(docker): fix image generation

rryter committed:
chore(logs): add fatal log

rryter committed:
chore(CI): use NX workflow

rryter committed:
chore(docker): fix image (tmp only, since it's much larger now)

rryter committed:
chore(build): fix it

rryter committed:
chore(DX): 1 command to start them all

rryter committed:
chore(nx): test build without cache

rryter committed:
chore(ci): install go

rryter committed:
chore(ci): install go

rryter committed:
chore(ci): install go

rryter committed:
chore(CI): increase logging

rryter committed:
chore(tests): adjust to latest changes, ready for writing actual tests

rryter committed:
chore(tests): created tests, ready to write them

rryter committed:
chore(ci): make sure go is installed on the agents

rryter committed:
chore(ci): enable cache again

rryter committed:
chore(cleanup): random stuff removed

rryter committed:
chore(readability): small improvements

rryter committed:
chore(errors): handle some more errors

rryter committed:
chore(css): use scss

rryter committed:
chore(docker): fix build

rryter commented:
Not sure if we should merge this in here, it' might be better to have this in a separate repository.
